Lib/test/test_concurrent_futures.py

```diff
@@ -1018,6 +1017,34 @@ def test_idle_process_reuse_multiple(self):
         self.assertLessEqual(len(executor._processes), 2)
         executor.shutdown()
 
+    def test_max_tasks_per_child(self):
+        executor = self.executor_type(1, max_tasks_per_child=3)
+        f1 = executor.submit(mul, 6, 7)
+        self.assertEqual(f1.result(), 42)
+        self.assertEqual(len(executor._processes), 1)
+        original_pid = [pid for pid in executor._processes.keys()][0]
+        f2 = executor.submit(mul, 6, 7)
+        f3 = executor.submit(mul, 6, 7)
+        self.assertEqual(f2.result(), 42)
+        self.assertEqual(f3.result(), 42)
```

Comment:
Why not assert that there weren't multiple PIDs up to this point?

Reply:
I can change around the order here and assert it on the 2nd result, but not the third.

Reply:
That's good. But why isn't the third run guaranteed to have the same PID?

Reply:
The problem is it's a race condition. 

If I submit the third job, I have to check `executor._processes` before that worker gets the work item, finishes it and shuts down. 

It could either result in an error doing `self.assertEqual(len(exectuor._processes), 1)` because the worker has already shut down. Even if that doesn't fail, there's no guarantee that the process hasn't already been replaced.

I could get around this by calling `sleep_and_print` but then it's just introducing a timing-dependent test and I figured we wouldn't want to do that.

Reply:
That makes sense. Thank you.